app/models/translation_engine/connection.rb

```diff
@@ -71,11 +71,15 @@ def get_translations_head
   private
 
   def connection(timeout = TranslationEngine.timeout)
-    Thread.current[:translation_server_connection] ||= Faraday.new(:url => TranslationEngine.api_host) do |faraday|
-      faraday.use TranslationEngine::ConnectionExceptionMiddleware
-      faraday.adapter Faraday.default_adapter
-      faraday.options.timeout      = timeout
-      faraday.options.open_timeout = TranslationEngine.timeout * 4
+    begin
```

Comment:
tohle se musi hodit kolem volani metody connection, tady to nic neudela, protoze do te doby, nez udelas call, tak tohle ti vytvori jen wrapper.